src/decoder.rs

```diff
@@ -109,24 +109,51 @@ fn decode_signature_struct(
     // Build signature structure to verify.
     let signature_bytes = &cose_signature[2];
     let signature_bytes = unpack!(Bytes, signature_bytes).clone();
+    let tmp = vec![];
     let sig_structure_bytes = get_sig_struct_bytes(
         protected_body_head.clone(),
         protected_signature_header_serialized.clone(),
+        &tmp, // no external_aad here.
         payload,
     );
 
     // Read intermediate certificates from protected_body_head.
-    let protected_body_head = unpack!(Bytes, protected_body_head);
-    let protected_body_head_map = match decode(protected_body_head) {
-        Ok(value) => value,
-        Err(_) => return Err(CoseError::DecodingFailure),
+    let protected_body_head = match protected_body_head.clone() {
+        CborType::Bytes(cbor_object) => cbor_object,
+        // A protected body head that is empty can be encoded as an empty map.
```

Comment:
Maybe add the explanation from the spec that tampering isn't possible because the verifier encodes the header, and the only way the signature will match is if the encodings match (if I have that right?)

Reply:
Added a small comment.